### ISSUE TITLE
Dual rule for initialize formals

### ISSUE THREAD

**Issue by alexeieleusis** [dart-lang/linter#485]
If a private field is initialized in the body of a constructor and could be replaced by a formal, a lint should be produce to move it to an initializer.

```dart
class MyClass {
    Foo _foo;

    MyClass.incorrect(Foo otherFoo) {
        _foo = otherFoo; // LINT
        ...
    }

    // This could also enable other lints like declare _foo as final or remove constructor body.
    MyClass.correct(Foo otherFoo) : _foo = otherFoo {
        ...
    }
}
```

**Comment by srawlins** [comment]
Kind of surprising that we don't have a rule yet for this; I guess it would be tricky to calculate whether the _value_ (the right side) is something accessible in initializers.
